src/dotty/tools/dotc/core/Names.scala

```diff
@@ -347,4 +347,27 @@ object Names {
       StringBuilder.newBuilder.mapResult(s => from.fromChars(s.toCharArray, 0, s.length))
     def apply(): Builder[Char, Name] = termNameBuilder
   }
+
+  implicit val NameOrdering: Ordering[Name] = new Ordering[Name] {
+    def compare(x: Name, y: Name): Int = {
+      if (x.isTermName && y.isTypeName) 1
+      else if (x.isTypeName && y.isTermName) -1
+      else if (x.start == y.start && x.length == y.length) 0
+      else {
+        val until = Math.min(x.length, y.length)
```

Comment:
The queston here was whether we want to write

     x.length.min(y.length)

or

    x.length min y.length

not go all the way back to `Math.min` :-)

I think `min` is a well-established operator, so infix is OK.